ignite/trainer/history.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+
+class History(object):
+    def __init__(self):
+        super(History, self).__init__()
+        self._history = []
+
+    def __iter__(self):
+        return iter(self._history)
+
+    def __setitem__(self, idx, value):
+        self._history[idx] = value
+
+    def __getitem__(self, idx):
+        return self._history[idx]
+
+    def __len__(self):
+        return len(self._history)
+
+    def clear(self):
+        self._history = []
+
+    def append(self, item):
+        self._history.append(item)
```

Comment:
If you only want to add a few methods to a regular Python list, inheritance might be simpler than composition (you don't have to redirect anything). The main difference is that `isinstance(History, list)` is `True`, which you might not want to have (although this can be fixed by implementing `__instancecheck__`).

Reply:
yeah, actually that is easier. I don't think having `isinstance(History, list)` being `True` is an issue (or not one that I can see), I'll update this to use inheritance